lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,14 +56,215 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHeadOrRefresh = (url, upstreamDocument, callback) ->
+  activeAssets = extractTrackedAssets(document)
+  upstreamAssets = extractTrackedAssets(upstreamDocument)
+  updateLinkTags(activeAssets, upstreamAssets)
+  updateScriptTagsOrRefresh(url, activeAssets, upstreamAssets, callback)
```

Comment:
@GoodForOneFare pointed this out earlier and I'm working on this part of the refactor right now, sorry for the delay.